L67-cpp-callback-api.md

```diff
@@ -0,0 +1,125 @@
+
+### Related proposals
+
+* The C++ callback API is built on top of a new [callback completion queue in core](https://github.com/grpc/proposal/pull/181)
+* The API structure has substantial similarities to the gRPC-Node and gRPC-Java APIs
+
+## Proposal
+
+The callback API is designed to have the performance and thread scalability of an asynchronous API without the burdensome programming model of the completion-queue-based model. In particular, the following are fundamental guiding principles of the API:
+
+* Library directly calls user-specified code at the completion of RPC actions.
+* No explicit polling required for notification of completion of RPC actions.
+   - In practice, these requirements mean that there must be a library-controlled poller for monitoring such actions.
+* As in the synchronous API, server RPCs have an application-defined method handler function as part of their service definition. The library invokes this method handler when a new server RPC starts.
+* Like the synchronous API and unlike the completion-queue-based asynchronous API, there is no need for the application to "request" new server RPCs. Server RPC context structures will be allocated and have their resources allocated as and when RPCs arrive at the server.
+
+### Reactor model
+
+The most general form of the callback API is built around a _reactor_ model. Each type of RPC has a reactor base class provided by the library. These types are:
+
+* `ClientUnaryReactor` for unary calls
+* `ClientBidiReactor` for bidi-streaming 
+* `ClientReadReactor` for server-streaming 
+* `ClientWriteReactor` for client-streaming 
+* `ServerUnaryReactor` for unary calls
+* `ServerBidiReactor` for bidi-streaming 
+* `ServerReadReactor` for client-streaming 
+* `ServerWriteReactor` for server-streaming 
+
+Client RPC invocations from a stub provide a reactor pointer as one of their arguments, and the method handler of a server RPC must return a reactor pointer.
+
+These base classes provide three types of methods:
+
+1. Operation-initiation methods: start an asynchronous activity in the RPC. These are methods provided by the class and are not virtual.
+   - `StartRead`, `StartWrite`, `StartWritesDone`: streaming operations as appropriate to the RPC variety. *Client and server*.
```

Comment:
Is `StartWritesDone()` useful on the server?  If so, what case does it cover?  When would an application want to use that instead of `StartWrites()`?

Reply:
Oops! That wasn't meant to be on the server.